docs/pages/example/hover-styles.html

```diff
@@ -35,27 +40,22 @@
         }
     });
 
-    map.addLayer({
-        "id": "state-fills-hover",
-        "type": "fill",
-        "source": "states",
-        "layout": {},
-        "paint": {
-            "fill-color": "#627BC1",
-            "fill-opacity": 1
-        },
-        "filter": ["==", "name", ""]
-    });
-
     // When the user moves their mouse over the states-fill layer, we'll update the filter in
     // the state-fills-hover layer to only show the matching state, thus making a hover effect.
     map.on("mousemove", "state-fills", function(e) {
-        map.setFilter("state-fills-hover", ["==", "name", e.features[0].properties.name]);
+        if (e.features.length > 0) {
+            if (hoveredStateId) {
+                map.setFeatureState({source: 'states', id: hoveredStateId}, { hover: false});
+            }
+            hoveredStateId = e.features[0].id;
+            map.setFeatureState({source: 'states', id: hoveredStateId}, { hover: true});
+        }
     });
 
     // Reset the state-fills-hover layer's filter when the mouse leaves the layer.
     map.on("mouseleave", "state-fills", function() {
-        map.setFilter("state-fills-hover", ["==", "name", ""]);
+        map.setFeatureState({source: 'states', id: hoveredStateId}, { hover: false});
```

Comment:
is `hoveredStateId` guaranteed to be defined? It seems like yes, but not 100% sure